python/ray/train/batch_predictor.py

```diff
@@ -66,6 +62,17 @@ def _predict_pandas(self, df, **kwargs) -> "pd.DataFrame":
             predictor_cls=PandasUDFPredictor,
         )
 
+    def get_preprocessor(self) -> Preprocessor:
+        if self._override_preprocessor:
+            return self._override_preprocessor
+
+        checkpoint_data = self._checkpoint.to_dict()
+        preprocessor = checkpoint_data.get(PREPROCESSOR_KEY)
```

Comment:
What’s the user journey that we are trying to solve by introducing these apis?

It seems by doing this we are making assumptions about the underlying checkpoint structure (assuming there is a PREPROCESSOR_KEY in the checkpoint). So these will only work for checkpoints outputted by our trainers and not for the general case. BatchPredictor will no longer work with any predictor and any checkpoint. I’m wondering if there is an alternative solution here to resolve the user problem.

Reply:
If it’s just the ability to override the preprocessor, then we currently have other mechanisms to do this and it might just be a matter of documenting them better

Reply:
(moved to slack)

Reply:
@matthewdeng this is the tricky issue

Reply:
Added `Checkpoint.get_preprocessor()` to unify this logic, though it's not optimized in this PR.